app/server/api/auth_api.py

```diff
@@ -342,7 +342,7 @@ def post(self):
                 user = User.query.filter_by(phone=phone).execution_options(show_all=True).first()
 
         # mobile user doesn't exist so default to creating a new wallet!
-        if user is None and phone:
+        if user is None and phone and current_app.config['ALLOW_SELF_SIGN_UP']:
```

Comment:
because this was defaulting to true, it was returning a `200` response for the first instance found, which was breaking the android app delegator